### ISSUE TITLE
Cannot convert java.util.Date

### ISSUE THREAD

**Issue by jean9696** [invertase/react-native-firebase#492]
### Issue

Hi, I'm trying to use firebase with react native on android with invertase/react-native-firebase but I have some dates in my firestoreDataBase and it stops my application everytime because of this error:
`Cannot convert java.util.Date`
I really don't know how to fix this since I do not know where this is handled in Java

My code to get my documents is as following 
```javascript
componentDidMount() {
    firebase.auth().signInAnonymously().then(() => {
      this.getEvents();
    });
  }

  getEvents = () => {
    this.unsubscribe();
    this.unsubscribe = firebase.firestore().collection('events').onSnapshot(eventsSnapshot => {
      this.setState({
        events: eventsSnapshot.docs,
      });
    });
  };
```

Thank you for your help !

### Environment

1. Application Target Platform: Android but both at the end

2. Development Operating System:  Windows 10

3. Build Tools: 

4. React Native version: ^0.48.4

5. RNFirebase Version: ^3.0.2

6. Firebase Module: Firestore and Auth

**Comment by jean9696** [comment]
I got a workaround with this in FirestoreSerialize.java line 200
```Java
        map.putString(key, value.toString());
        // throw new RuntimeException("Cannot convert object of type " + valueClass);
```
But I'm sure that there is a proper way to do this !

**Comment by Ehesp** [comment]
Hey thanks for reporting. @chrisbianca is currently working on the data type conversions and believe this one is todo.

**Comment by chrisbianca** [comment]
@jean9696 This is actually fixed, pending a release which we'll hopefully get out at some point today.  Just trying to get another couple of bits sorted at the same time.

**Comment by chrisbianca** [comment]
This is now released as part v3.0.3

**Comment by jean9696** [comment]
@chrisbianca Wow thank you for your reactivity !! 👍
